assets/model_monitoring/components/src/model_monitor_evaluate_metrics_threshold/evaluate_metrics_threshold.py

```diff
@@ -27,12 +27,21 @@ def evaluate_metrics_threshold(
     is_nan_metrics_threshold_df = metrics_to_evaluate_df.filter(
         metrics_to_evaluate_df.threshold_value.isNull()
     )
+
+    is_nan_metrics_threshold_df = is_nan_metrics_threshold_df.filter(
+        is_nan_metrics_threshold_df.metric_value.isNull()
+    )
+
     metrics_without_threshold_count = is_nan_metrics_threshold_df.count()
 
     is_not_nan_metrics_threshold_df = metrics_to_evaluate_df.filter(
         metrics_to_evaluate_df.threshold_value.isNotNull()
     )
 
+    is_not_nan_metrics_threshold_df = is_not_nan_metrics_threshold_df.filter(
+        is_not_nan_metrics_threshold_df.metric_value.isNotNull()
+    )
+
     is_not_nan_metrics_threshold_df = is_not_nan_metrics_threshold_df.where(
         F.col("metric_value") > F.col("threshold_value")
     )
```

Comment:
Could you please also update the line 49-50? I think the logic here is trying to `show()` all the rows, hence the union. With your change it seems some records have been dropped.
Another option is to remove these 2 lines and make it something like:
```metrics_threshold_breached_df = is_not_nan_metrics_threshold_df.where(
        F.col("metric_value") > F.col("threshold_value")
    )
    
if metrics_threshold_breached_df .count() > 0:
    ```

Reply:
Please confirm with Mathieu whether we'd like to "show" the new metrics in the output (1st option), or don't show anything (2nd option) or keep showing the other metrics but not the new one, which IMO is kind of weird (keep it as-is).

Reply:
<img width="522" alt="image" src="https://github.com/Azure/azureml-assets/assets/42190563/1dd4c89d-fd5c-414a-9de0-35e94fe495b6">